### PR TITLE
Add modify-data command to merge datasets

### PR DESCRIPTION
**Pull request by FelixBoehme**

### CODE REVIEW

File: lsr_benchmark/_commands/_modify_data.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            current_offset += len(data)
+
+    return {
+        "data": np.concatenate(all_data),
+        "indices": np.concatenate(all_indices),
+        "indptr": np.concatenate(all_indptr),
+    }
+
+
+def perform_dataset_join(dataset: str, tira: Client, tira_dir: str) -> Path:
```
Comment (mam10eks) [inline lsr_benchmark/_commands/_modify_data.py:150]: Is this dataset join needed, or can it be removed? I think it would be a bit easier when we only join/merge embeddings?

File: lsr_benchmark/_commands/_modify_data.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    data_dir: str,
+) -> dict[str, np.ndarray]:
+    all_data = []
+    all_indices = []
+    all_indptr = []
+    current_offset = 0
+
+    for i, emb_path in enumerate(tqdm(embedding_paths, desc=f"Loading {data_dir}", leave=False)):
+        with np.load(emb_path / data_dir) as npz:
+            data = npz["data"]
```
Comment (mam10eks) [inline lsr_benchmark/_commands/_modify_data.py:121]: What is with the case that document IDs could be the same? The handling of duplicates should differ between queries (there, we should "scope" the Query Ids, so if one dataset has query-1, and another merged dataset also has query-1, we then want both. This is not true for documents.

For documents in the merges above, we would skip duplicates.

Comment (mam10eks) [reply lsr_benchmark/_commands/_modify_data.py:121]: Can you please add a unit test to ensure this duplicate handling is done as expected? Thanks in advance :)

File: lsr_benchmark/_commands/_modify_data.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    tira_dir = default_tira_cache_dir()
+
+    created_dataset_dirs = []
+    created_embedding_dirs = []
+
+    if join:
+        for dataset in tqdm(datasets, desc="Joining"):
+            created_dataset_dirs.append(perform_dataset_join(dataset, tira, tira_dir))
+            for emb in tqdm(embedding, desc="Processing Embeddings"):
+                created_embedding_dirs.append(perform_embedding_join(dataset, emb, tira, tira_dir))
```
Comment (mam10eks) [inline lsr_benchmark/_commands/_modify_data.py:297]: Is ensured that only successfull joins are persisted? E.g., first write to a temporary directory, then move it to the result directory?

File: lsr_benchmark/_commands/test_modify_data_cli.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import io
+import json
+
+import numpy as np
+
+from ._modify_data import load_and_merge_embeddings, prefix_json
+
+
+def test_load_and_merge_embeddings(tmp_path):
```
Comment (mam10eks) [inline lsr_benchmark/_commands/test_modify_data_cli.py:10]: I think it would make sense to add a few more small test cases, as described above, to verify the different expected behaviour of duplicates in queries and in documents.

**Review by mam10eks — CHANGES_REQUESTED** [review]
Looks in most cases good to me, only a few minor things that I shortly described here need to be resolved. I explain the things a bit in more detail in a private chat :)

File: lsr_benchmark/_commands/_modify_data.py (hunk 1 above)
Comment (FelixBoehme) [reply lsr_benchmark/_commands/_modify_data.py:150]: I would have thought it to be necessary when we want to use a lexical approach like bm25 on the joined datasets

File: lsr_benchmark/_commands/_modify_data.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+JOINT_TO_DATASETS = {
+    "msmarco-passage-trec-dl-2019+2020-judged": {
+        "settings": DuplicateHandling(docs=DuplicateBehaviour.SKIP),
+        "datasets": [
+            "trec-28-deep-learning-passages-20250926-training",
+            "trec-29-deep-learning-passages-20250926-training",
+        ],
+    },
+    "disks45-nocr-trec-robust-2004-fold1+2+3+4+5": {
+        "settings": DuplicateHandling(),
```
Comment (mam10eks) [inline lsr_benchmark/_commands/_modify_data.py]: Here docs=DuplicateBehaviour.SKIP makes sense

File: lsr_benchmark/_commands/_modify_data.py [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+        "datasets": [
+            "trec-robust-2004-fold-1-20250927-test",
+            "trec-robust-2004-fold-2-20250926-test",
+            "trec-robust-2004-fold-3-20250926-test",
+            "trec-robust-2004-fold-4-20250926-test",
+            "trec-robust-2004-fold-5-20250926-test",
+        ],
+    },
+    "clueweb12-trec-web-2013+2014+clueweb12-b13-trec-misinfo-2019": {
+        "settings": DuplicateHandling(),
```
Comment (mam10eks) [inline lsr_benchmark/_commands/_modify_data.py]: Here this makes sense: docs=DuplicateBehaviour.SKIP, queries=Prefix

File: lsr_benchmark/_commands/_modify_data.py [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+        "datasets": [
+            "trec-22-web-20251008-test",
+            "trec-23-web-20251008-test",
+            "trec-28-misinfo-20251008_1-test",
+        ],
+    },
+    "clueweb09-en-trec-web-2009+2010+2011+2012": {
+        "settings": DuplicateHandling(),
```
Comment (mam10eks) [inline lsr_benchmark/_commands/_modify_data.py]: Here: docs=DuplicateBehaviour.SKIP

File: lsr_benchmark/_commands/_modify_data.py (hunk 3 above)
Comment (mam10eks) [reply lsr_benchmark/_commands/_modify_data.py:297]: Is this already resolved?

**Review by mam10eks — CHANGES_REQUESTED** [review]
Looks almost good, only few minor changes, then we can merge :)

**Review by mam10eks — APPROVED** [review]

**Comment by mam10eks** [comment]
Looks all good, thanks!
